flit/wheel.py

```diff
@@ -156,12 +163,120 @@ def post_build(self):
 
         if self.upload:
             from .upload import do_upload
-            do_upload(self.wheel_file, self.metadata, self.repo)
+            do_upload(self.wheel_path, self.metadata, self.repo)
 
     def build(self):
-        self.clean_build_dir()
         self.copy_module()
         self.write_metadata()
         self.write_record()
-        self.zipup()
+
+        self.wheel_zip.close()
+
         self.post_build()
+
+
+import stat, time
+from zipfile import ZipInfo, ZIP_LZMA, _get_compressor, ZIP64_LIMIT, crc32
+
+def my_zip_write(self, filename, arcname=None, compress_type=None,
```

Comment:
It's actually now a bit of a hodgepodge of code from Python 3.4 and 3.5, because neither version would quite work with the other. Hopefully by 3.6 we can add functionality to `zipfile` so this is no longer necessary.

Nice! I noticed I'd got another ~20 stars. Flit is slowly closing on Pynsist as my most starred project.